fuse_variables/CMakeLists.txt

```diff
@@ -1,35 +1,28 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.16)
 project(fuse_variables)
 
-set(build_depends
-  fuse_core
-  pluginlib
-  roscpp
-)
+# Default to C++17
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CXX_STANDARD_REQUIRED YES)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  ${build_depends}
-)
-find_package(Ceres REQUIRED)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  LIBRARIES
-    ${PROJECT_NAME}
-  CATKIN_DEPENDS
-    ${build_depends}
-  DEPENDS
-    CERES
-)
+find_package(ament_cmake REQUIRED)
+find_package(fuse_core REQUIRED)
+find_package(pluginlib REQUIRED)
+
+find_package(Ceres REQUIRED)
 
 ###########
 ## Build ##
 ###########
-add_compile_options(-Wall -Werror)
 
 ## fuse_variables library
-add_library(${PROJECT_NAME}
+add_library(${PROJECT_NAME} SHARED
```

Comment:
same comment about leaving off `SHARED` and using `ament_cmake_ros` instead of `ament_cmake` https://github.com/locusrobotics/fuse/pull/287#discussion_r1028631887